nutree/typed_tree.py

```diff
@@ -266,7 +267,11 @@ def add_child(
                 # raise NotImplementedError("Cross-tree adding")
             if data_id and data_id != source_node._data_id:
                 raise UniqueConstraintError(f"data_id conflict: {source_node}")
-            node = factory(
+
+            # If creating an inherited node, use the parent class as constructor
+            child_class = child.__class__
```

Comment:
should the `factory` argument take precendence, when passed? (of course then line 253 must not override it)